packages/mds-compliance-engine/engine/mds-compliance-engine.ts

```diff
@@ -86,6 +86,12 @@ export function processPolicy(
   if (isPolicyActive(policy)) {
     const provider_ids = getProviderIDs(policy.provider_ids)
     const results = provider_ids.map(provider_id => {
+      /**
+       * If there's no data for this provider, skip it.
+       * This allows querying for a given set of providers at a time, instead of _all_ providers.
+       */
+      if (!providerInputs.hasOwnProperty(provider_id)) return undefined
```

Comment:
But `providerInputs` is always going to have the `provider_id`, isn't it? The return statement of `getProviderInputs` is `    return { filteredEvents, deviceMap, provider_id }`, and `provider_id` is a parameter passed in to that helper.

Reply:
Negative, this isn't checking to see if there's a property called `'provider_id'` in the object. The `ProviderInputs` type is an object which'd look like, for example:
```
{ 
  '30360003-90c1-44f4-8d50-20debcb8fbd5': { filteredEvents, deviceMap, provider_id },
  '09dc8093-7b4f-430f-a6e3-c9d873c2adce': { filteredEvents, deviceMap, provider_id }
  }
 ```
  
In this case, we're checking to see if that object even has data for that provider. In the context of the batch processor, it will only have data for one provider at a time now.
At runtime, this could look something like `if (!providerInputs.hasOwnProperty('09dc8093-7b4f-430f-a6e3-c9d873c2adce') return undefined`